students/uss_tomasz/lesson_02_flow_control/main.py

```diff
@@ -0,0 +1,165 @@
+x_finish = int(input("Finishing Column of king movement"))
+y_finish = int(input("Finishing Row of king movement"))
+
+if abs(x_start - x_finish) < 2 and abs(y_start - y_finish) < 2:
+    print("YES")
+else:
+    print("NO")
+
+# Problem Bishop moves
+x_start = int(input("Starting Column of bishop movement"))
+y_start = int(input("Starting Row of bishop movement"))
+x_finish = int(input("Finishing Column of bishop movement"))
+y_finish = int(input("Finishing Row of bishop movement"))
+
+if abs(x_finish - x_start) == abs(y_finish - y_start):
+    print("YES")
+else:
+    print("NO")
+
+# Problem Queen move
+x_start = int(input("Starting Column of queen movement"))
+y_start = int(input("Starting Row of queen movement"))
+x_finish = int(input("Finishing Column of queen movement"))
+y_finish = int(input("Finishing Row of queen movement"))
+
+if (abs(x_finish - x_start) == abs(y_finish - y_start)) or (x_start == x_finish or y_start == y_finish):
+    print("YES")
+else:
+    print("NO")
+
+# Problem Knight move
+x_start = int(input("Starting Column of knight movement"))
+y_start = int(input("Starting Row of knight movement"))
+x_finish = int(input("Finishing Column of knight movement"))
+y_finish = int(input("Finishing Row of knight movement"))
+
+if (abs(x_finish - x_start) == 2 and abs(y_finish - y_start) == 1) or \
+        (abs(x_finish - x_start) == 1 and abs(y_finish - y_start) == 2):
+    print("YES")
+else:
+    print("NO")
+
+# Problem Chocolate bar
+n = int(input("Chocolate bar - Portions height"))
+m = int(input("Chocolate bar - Portions length"))
+k = int(input("Chocolate bar - Desired squares"))
+if n * m < k:
+    print("NO")
+else:
+    if k % n == 0 or k % m == 0:
+        print("YES")
+    else:
+        print("NO")
+
+# Problem Leap year
+year = int(input("Year to be checked"))
+if year % 4 == 0 and year % 100 != 0 or year % 400 == 0:
+    print("LEAP")
+else:
+    print("COMMON")
+
+'''
+https://snakify.org/lessons/for_loop_range/problems/
+'''
+# Problem Factorial
+n = int(input("Number for factorial calculation"))
+factorial = 1
+for i in range(1, n + 1):
+    factorial *= i
+print("Factorial is ", factorial)
+
+# Problem The number of zeros
+quantity = int(input("How many numbers do you want to check?"))
+zeros_counter = 0
+for i in range(quantity):
+    variable = int(input("Number to check is it 0"))
+    if variable == 0:
+        zeros_counter += 1
+print("Zero counted: ", zeros_counter)
+
+# Problem Adding factorials
+n = int(input("Number to check sum of factorials"))
+factorial_sum = n
+for i in range(n - 1, 0, -1):
+    factorial_sum += 1
+    factorial_sum *= i
+print("Total sum of following factorials is ", factorial_sum)
+
+# Problem Ladder
+n = int(input("Ladder steps [between 1 and 9] = "))
+for k in range(1, n + 1):
+    for j in range(1, k + 1):
+        print(j, end='')
+    print()
+
+# Problem Lost card
+N = int(input("Cards quantity = "))
+table = [0 for i in range(N)]
```

Comment:
`0` is a valid int, use `None` if you want to mark missing/invalid/not initialized value
```
> [None]*6
[None, None, None, None, None, None]
```